src/cpp/session/modules/SessionDataViewer.R

```diff
@@ -890,3 +904,8 @@
    invisible("")
 })
 
+.rs.addFunction("isNonEmptyScalarString", function(x)
+{
+   is.character(x) && length(x) == 1 && nzchar(x)
```

Comment:
Kevin enlightened me that all objects have length 1, so even `length('hello') == 1`.

So `length(x) == 1` in this case just checks that it isn't `NULL`, because `length(NULL) == 0`

Reply:
`cacheKey` should be a character vector of length one, whose string is non-empty. So, a normal string like `"hello"`, but not `""` or `c("a", "b")`.